src/coreComponents/common/MpiWrapper.hpp

```diff
@@ -267,7 +258,7 @@ struct MpiWrapper
    * @return MPI_SUCCESS or and MPI_ERROR from internal calls to MPI_WaitAny.
    */
   static int activeWaitSomeCompletePhase( const int participants,
-                                          std::vector< std::function< MPI_Request ( int ) > > const & phases );
+                                          std::vector< std::tuple< MPI_Request *, MPI_Status *, std::function< void ( int ) > > > const & phases );
```

Comment:
Maybe worth introducing a simple struct with named fields to replace tuple?

Reply:
Yeah...I think we need something a little more structured in general. I want to re-organize the entirety async communication utilities.

Reply:
....or to be precise, I want @wrtobin to re-organize it :)